packages/lesswrong/server/authenticationMiddlewares.ts

```diff
@@ -178,14 +232,47 @@ export const addAuthMiddlewares = (addConnectHandler) => {
     })
   }
   
+
+  // NB: You must also set the expressSessionSecret setting
+  const auth0ClientId = auth0ClientIdSetting.get();
+  const auth0OAuthSecret = auth0OAuthSecretSetting.get()
+  const auth0Domain = auth0DomainSetting.get()
+  if (auth0ClientId && auth0OAuthSecret && auth0Domain) {
+    passport.use(new Auth0Strategy(
+      {
+        clientID: auth0ClientId,
+        clientSecret: auth0OAuthSecret,
+        domain: auth0Domain,
+        callbackURL: `${getSiteUrl()}auth/auth0/callback`,
```

Comment:
We briefly discussed using something like [url-join](https://www.npmjs.com/package/url-join) for this kind of thing. Might not be worth it, but also might be good?